kotlinx-coroutines-core/jvm/src/DefaultExecutor.kt

```diff
@@ -4,10 +4,25 @@
 
 package kotlinx.coroutines
 
+import kotlinx.coroutines.internal.*
 import java.util.concurrent.*
 import kotlin.coroutines.*
 
-internal actual val DefaultDelay: Delay = DefaultExecutor
+internal actual val DefaultDelay: Delay = initializeDefaultDelay()
+
+private val defaultMainDelayOptIn = systemProp("kotlinx.coroutines.main.delay", true)
```

Comment:
I think this should be documented at least somewhere.

Reply:
We can mention it in the original issue and in the changelog

Reply:
Also, we can document `Delay` itself, but it doesn't make it to the Dokka and is internal interface, so it's unlikely someone will read it

Reply:
That depends. Do we plan to remove this opt-in flag in a release or two if no one complains or is it likely that we'll keep it? If it's the latter, then I don't think only documenting a configuration parameter in a changelog (or a closed issue) is sufficient.

Reply:
I do not see any urge in removing it (e.g. we're still preserving flag for fast service loader), its maintenance has basically zero cost

Reply:
A suggestion then: maybe later, outside of this pull request, we should document in the README (either the top-level one or for the core module) all the flags that our code non-transitively recognizes (or even transitively, if viable).

Reply:
I would advise against it. All the options are intended to be user-facing features, but rather emergency brakes when everything went terribly wrong. Some of the flags (e.g. in the scheduler) can only be used for specific performance scenarios, and the overall behaviour with an arbitrary set of options is not well-defined.